# Probability

Probability is a branch of maths that allows us to quantify uncertainty

## Set theory

In simple terms, a set is a collection of things. For example, we can use a set to represent items in a backpack. We might have: 

{Book,Paper,Folder,Hat,Pen,Snack}

Sets are notated with curly braces and follow two rules – elements are unique and are in no particular order. Therefore: {1,2,3,4,5} = {5,3,2,4,1}

Sets are often defined with a capital letter:

A = {1,2,3,4,5}

Sets can also contain subsets. Set A is a subset of set B if all the elements in A exist within B. For example:
 
A = {1,2,3}
B = {1,2,3,4,5}
​	 
Here, set A is a subset of B because all elements of A are contained within B.

### Experiments and sample spaces

In probability, an experiment is something that produces observation(s) with some level of uncertainty. A sample point is a single possible outcome of an experiment. A sample space is the set of all possible sample points for an experiment.

For example, if we flip a coin twice and record whether each flip results in heads or tails, there would be four sample points in t1he experiment: two heads (HH), tails and then heads (TH), heads and then tails (HT), or two tails (TT). The sample space can be written as:

S = {HH,TT,HT,TH}

A specific outcome or set of outcomes is an event (E), and a subset of the sample space. Events could include: 

Getting Two Heads
A={HH}

Getting Two Tails
B={TT}

Getting Both a Heads and Tails
C={HT,TH}

## Calculating probability 
​
Probability can be defined as "If we run an experiment an infinite amount of times, the probability of each event is the proportion of times it occurs."

The probability of an event happening is the number of times an event occured divided by the number of trials.

If we ran 1000 coin flip experiments and we got two heads 255 times, the probability would be:


In [3]:
P = 255 / 1000
P

0.255

This means there's an estimated 25.5% chance of getting two heads when flipping coins.

It's not feasible often to run an infinite number of experiments to find the true probability, but the law of large numbers states that as the experiment is run more and more, the observed proportion of times each event occurs will converge to its true probability. 

## Union, intersection and complement

<img src="Assets/union.jpg">

If we have two sets, A and B. A represents rolling an odd number with a 6-side dice {1,3,5}, and B represents rolling a number greater than two {3,4,5,6}.

### Union 

The union of two sets encompasses any element that exists in either one or both of them. The union of set A and B is {1,3,4,5,6}. Union can be written as (A or B).

Addition rule formula

P(A or B) = P(A) + P(B) - P(A and B)

### Intersection 

Intersesction encompasses any element that exists in both sets. Using the same sets as above, the intersection of A and B would be {3,5}. Intersection is written as (A and B).

The complement of a set consists of all possible outcomes outside the set. In this example, the complement of A would be rolling an even 4number {2,4,6}. The complement of A is written as A𝘤. Together the set and its complement cover the entire sample space.


### Mutually exclusive 

Mutually exclusive events are ones that can't happen at the same time. For example, it's not possible to get both heads and tails from a single coin flip.

## Dependent vs. independent events

Independent events are ones that do not effect future ones. A coin flip does not effect later flips.

Dependent events are ones where a previous event does effect future events. If a bag has five red marbles and three green marbles, the probability of taking a green marble out of the bag in the second event depends on the colour of the marble in the first event.

## Addition rule

Event A is rolling an odd number on a six sided dice, event B is rolling a number greater than two. To find the probability of one or both events
P(A or B)= P(A)+P(B)−P(A and B)

Mutually exclusive